feat: add network isolation modes to container/run.sh script

keivenchang opened the pull request:
#### Overview:

Add network isolation modes to container run script with comprehensive configuration options. This enables secure CI/CD pipelines and flexible networking for different deployment scenarios.

#### Details:

- Add `--network` flag to `run.sh` supporting host, bridge, none, and container sharing modes
- Update container README with detailed network configuration documentation and use cases
- Add CI/CD workflow examples demonstrating bridge networking for test isolation
- Document performance/security tradeoffs and limitations for each network mode
- Include comparison table and warnings for restricted functionality modes

#### Where should the reviewer start?

- `container/run.sh` - New network flag implementation and argument parsing
- `container/README.md` - Comprehensive network configuration documentation and examples

/coderabbit profile chill

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added a configurable networking option to the container runner with modes: host (default), bridge, none, and container sharing. Help text updated with usage and examples.

* **Documentation**
  * Expanded network configuration guidance with detailed use cases and examples for each mode, including custom networks.
  * Updated usage examples to be network-aware; removed outdated GPU-focused example.
  * Introduced a CI/CD Workflow section emphasizing network isolation and in-container integration tests.
  * Augmented examples for workspace mounting, volume/cache flags, and a multi-service startup sequence.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

keivenchang committed:
feat: add network isolation modes to container run script

- Add --network flag to run.sh supporting host, bridge, none, and container sharing
- Update README with comprehensive network configuration documentation
- Add CI/CD workflow examples with bridge networking for isolation
- Document limitations and use cases for each network mode

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
docs: clarify port sharing limitations and improve examples

- Add warning about host networking port sharing (only one NATS/etcd instance)
- Update network mode comparison table with Port Sharing column
- Fix production workflow to use non-local-dev image (runs as root)
- Improve CI/CD examples to use appropriate image types
- Clarify bridge networking setup instructions for full stack testing

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

kthui reviewed (approved):
Nice! We have network isolation to the container.

One thing people might ask in the future is adding port publish
```
-p HOST_PORT:CONTAINER_PORT
```
because the host or other containers no longer have access to network services at the container if bridge mode is used, so network service ports will need to be explicitly published, but I think this can be done in a separate PR given the default option is still "host".

keivenchang committed:
docs: enhance container README with comprehensive GPU config and CI/CD workflow

- Add detailed port sharing limitations and warnings for host networking
- Restructure CI/CD workflow into clear numbered steps
- Add comprehensive framework-specific GPU memory configuration
- Include inline argument explanations for vLLM, SGLang, and TensorRT-LLM
- Improve code block formatting with proper bash syntax highlighting
- Remove redundant sections and streamline documentation flow
- Add bridge networking isolation benefits and setup instructions

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
docs: standardize CI/CD workflow format and remove framework bias

- Unify CI/CD workflow format to match Development/Production workflows
- Use single code block with numbered comments for consistency
- Remove framework bias by treating vLLM, SGLang, TensorRT-LLM equally
- Uncomment all framework alternatives for user choice
- Consolidate framework-specific arguments into compact summary
- Improve documentation consistency across all workflow sections

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
feat: add port mapping support and enhance container documentation

- Add --port|-p option to run.sh for Docker port mapping
- Integrate PORT_MAPPINGS variable into docker run command
- Update help text with port mapping documentation
- Add cache volume mount (-v /home/ubuntu/.cache:/home/ubuntu/.cache) to all examples
- Simplify bridge networking examples and remove redundant port examples
- Add port sharing note with host_port:container_port format clarification
- Enhance network mode comparison table with port publishing column
- Standardize container examples across all networking modes

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang commented:
> Nice! We have network isolation to the container.
> 
> One thing people might ask in the future is adding port publish
> 
> ```
> -p HOST_PORT:CONTAINER_PORT
> ```
> 
> because the host or other containers no longer have access to network services at the container if bridge mode is used, so network service ports will need to be explicitly published, but I think this can be done in a separate PR given the default option is still "host".

Ok, I just added the --port|-p option and mentioned in the README.md about it.
